Makefile

```diff
@@ -45,13 +45,13 @@ docker-offsets:
 	docker run --rm -v $(shell pwd):/app golang:1.20 /bin/sh -c "cd ../app && make offsets"
 
 .PHONY: update-licenses
-update-licenses: | $(GOLICENSES)
+update-licenses: | generate $(GOLICENSES)
```

Comment:
I do not understand why we need to depend on `generate`. The generated files should be committed to the repository. _This should be addressed in a separate PR._

Reply:
is it possible that the re-generated files cause an update to LICENSES?

Reply:
I couldn't get the thing to pass without first calling `make generate`. Then I noticed that `build` had a dependency on generate as well.  Since the tool essentially builds the app when its rummaging through the packages I added `generate` as a dependency as well.

I don't have a good understanding of all the things `cli/main.go` requires to be built.  It is possible go-licenses only needs the CFLAGS env var, but I know that calling `generate` first allows `go-licenses` to run as expected.

Reply:
I tests with only the env var again and it isn't enough.  We need the output from `make generate` in order for `go-licenses` to be able to handle the files that depend on the bpf files.

Reply:
Ah yeah, the `*.o` files are the ebpf programs. these are essentially compiled machine code programs that we shouldn't commit

Reply:
see https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/16#issuecomment-1318259874